src/Arrays/MPIStateArrays.jl

```diff
@@ -856,4 +858,44 @@ end
     end
 end
 
+"""
+    throw_isnan_error(Q::MPIStateArray)
+
+Call if `isnan(norm(Q)) == true` to provide a
+detailed error of which field(s) have NaNs.
+
+!!! note
+    This method assumes that `isnan(norm(Q)) == true`,
+    and throws an error otherwise.
```

Comment:
Why? Call this something like `throw_on_isnan` and let it be general.

Reply:
`norm` is expensive, and my thought was that we don't need to include unnecessary calls of `norm(Q)` if we assume that things are already determined to be broken.

Reply:
We can assume that it's been called without forcing it -- just document that it's expensive and this is how it _should_ be called and remove the defensive check below? But I'm not insisting...

Reply:
Looped in some others for review.

Reply:
I've thought about this more and I think you're right. It's already an expensive function and we might as well just build in the assumption and document that. Updating...